students/perrymant/22-Logging.md

```diff
@@ -0,0 +1,124 @@
+| Alert         | Action must be taken immediately  | A condition that should be corrected immediately, such as a corrupted system database. |
+| Critical      | Critical conditions               | Hard device errors.                                                                    |
+| Error         | Error conditions                  |                                                                                        |
+| Warning       | Warning conditions                |                                                                                        |
+| Notice        | Normal but significant conditions | Conditions that are not error conditions, but that may require special handling.       |
+| Informational | Informational messages            |                                                                                        |
+| Debug         | Debug-level messages              | Messages that contain information normally of use only when debugging a program.       |
+(source [Wiki](https://en.wikipedia.org/wiki/Syslog))
+
+- In Spring Boot Logging Levels are: `OFF`, `ERROR`, `WARN`, `INFO`, `DEBUG`, `TRACE`.
+- By default, Spring Boot configures logging via [Logback](http://logback.qos.ch) to write to the console at an INFO level.
+
+### Common log libraries in Java
+- `SLF4J` (Simple Logging Facade for Java, a facade is an object that serves as a front-facing interface masking more complex underlying or structural code. Here's a `HelloWorld` using `SLF4J`:
+```java
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HelloWorld {
+  public static void main(String[] args) {
+    Logger logger = LoggerFactory.getLogger(HelloWorld.class);
+    logger.info("Hello World");
+  }
+}
+```
+This would result in the following log message: `0 [main] INFO HelloWorld - Hello World`
+
+- Here's an example with `log4j`, logging in 3 different log levels:
+```java
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+
+public class Log4jExample {
+
+    private static Logger logger = LogManager.getLogger(Log4jExample.class);
+
+    public static void main(String[] args) {
+        logger.debug("Debug log message");
+        logger.info("Info log message");
+        logger.error("Error log message");
+    }
+}
+```
+This would result in the following log message:
+```
+15:11:22.022 [main] DEBUG Log4jExample - Debug log message
+15:11:22.027 [main] INFO Log4jExample - Info log message
+15:11:22.027 [main] ERROR Log4jExample - Error log message
+```
+
+For full control over the logging configuration in Intellij, you can create a logback.xml file at the root of the classpath (in src/main/resources). Here’s an example of a simple log-back.xml file you might use (this Logback configuration is more or less equivalent to the default you’ll get if you have no logback.xml file.):
```

Comment:
This is unrelated to IntelliJ. It's also not really the root of the class path, so I'd just say `src/main/resources` and don't mention the classpath.